### PR TITLE
NET-407 Update S2551 rule description.

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Fixes [NET-407](https://sonarsource.atlassian.net/browse/NET-407)
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

[NET-407]: https://sonarsource.atlassian.net/browse/NET-407?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: rules/S2551/csharp/rule.adoc [hunk 1]
```diff
@@ -1,10 +1,8 @@
-include::../why-dotnet.adoc[]
-
-For example, a `string` should never be used for locking. When a `string` is https://en.wikipedia.org/wiki/Interning_(computer_science)[interned] by the runtime, it can be shared by multiple threads, breaking the locking mechanism.
```
Comment (martin-strecker-sonarsource) [inline rules/S2551/csharp/rule.adoc:3]: This sentence was moved to the common description (why-dotnet)

File: rules/S2551/csharp/rule.adoc [hunk 2]
```diff
@@ -1,10 +1,8 @@
...
+The instance passed to the https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock[`lock` statement] should be a dedicated private field.
 
-Instead, a dedicated private `object` instance should be used for each shared resource. This minimizes access to the lock instance, avoiding deadlocks and lock contention.
```
Comment (martin-strecker-sonarsource) [inline rules/S2551/csharp/rule.adoc:5]: Also moved to "why-dotnet"

File: rules/S2551/csharp/rule.adoc [hunk 3]
```diff
@@ -45,4 +47,7 @@ void MyLockingMethod()
 
 include::../resources-dotnet.adoc[]
 
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock[The lock statement - ensure exclusive access to a shared resource]
```
Comment (martin-strecker-sonarsource) [inline rules/S2551/csharp/rule.adoc:50]: I hope this works. The idea is to append a dedicated C# link to the common "Documents" bullet list.

File: rules/S2551/csharp/rule.adoc [hunk 4]
```diff
@@ -1,10 +1,8 @@
-include::../why-dotnet.adoc[]
-
-For example, a `string` should never be used for locking. When a `string` is https://en.wikipedia.org/wiki/Interning_(computer_science)[interned] by the runtime, it can be shared by multiple threads, breaking the locking mechanism.
+The instance passed to the https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock[`lock` statement] should be a dedicated private field.
```
Comment (martin-strecker-sonarsource) [inline rules/S2551/csharp/rule.adoc:1]: It is odd that the rule never mentioned the `lock` statement or linked to it. The statement is essential to the rule. I fixed this by adding a [Short description (no title)](https://github.com/SonarSource/rspec/blob/master/docs/description.adoc#2-learn-as-you-code-rule-format) with the link.

File: rules/S2551/csharp/rule.adoc (hunk 3 above)
Comment (CristianAmbrosini) [reply rules/S2551/csharp/rule.adoc:50]: Not sure if the empty line on top allows you to do that, let's see

File: rules/S2551/csharp/rule.adoc [hunk 5]
```diff
@@ -45,4 +47,7 @@ void MyLockingMethod()
 
 include::../resources-dotnet.adoc[]
 
+* Microsoft Learn - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock[The lock statement - ensure exclusive access to a shared resource]
+
+
```
Comment (CristianAmbrosini) [inline rules/S2551/csharp/rule.adoc:52]: Empty line

**Review by CristianAmbrosini — APPROVED** [review]
LGTM!

**Pull request merged**